src/main/java/ngrok/configuration/NgrokAutoConfiguration.java

```diff
@@ -1,24 +1,22 @@
 package ngrok.configuration;
 
 import lombok.extern.slf4j.Slf4j;
+import ngrok.NgrokComponent;
 import ngrok.NgrokRunner;
 import ngrok.api.NgrokApiClient;
 import ngrok.os.NgrokBinaryProvider;
 import ngrok.os.NgrokPlatformDetector;
 import ngrok.os.NgrokSystemCommandExecutor;
 import ngrok.util.NgrokDownloader;
 import org.springframework.beans.factory.annotation.Qualifier;
-import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
 import org.springframework.context.ApplicationEventPublisher;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.ComponentScan;
-import org.springframework.context.annotation.Configuration;
 import org.springframework.core.task.TaskExecutor;
 
 @Slf4j
 @ComponentScan(basePackages = "ngrok")
-@Configuration
-@ConditionalOnBean(NgrokConfiguration.class)
+@NgrokComponent
```

Comment:
I use `@NgrokComponent` to to handle "enabling and disabling" ngrok related beans, and here is a test for that -> https://github.com/kilmajster/ngrok-spring-boot-starter/blob/main/src/test/java/ngrok/configuration/NgrokAutoConfigurationTest.java in all other tests there is `ngrok.enable=true` property included. I just wanted to handle testing of autoconfiguration in generic way somehow :)

Reply:
Ok, I see :) That seems to be fine. Maybe having a look into how other starters handle this could be a good idea. I don't know whats the best solution there too 😉